### PR TITLE
Modify Corpus to store module specs as a tuple

### PR DESCRIPTION
**Pull request by Northbadge**
Part of #96

### CODE REVIEW

File: compiler_opt/rl/corpus.py [hunk 1]
```diff
@@ -122,6 +122,10 @@ def filter(self, p: re.Pattern):
     """Filters module specs, keeping those which match the provided pattern."""
     self._module_specs = [ms for ms in self._module_specs if p.match(ms.name)]
 
+  @property
+  def modules(self):
+    return list(self._module_specs)
```
Comment (mtrofin) [inline compiler_opt/rl/corpus.py:127]: why constrain as list?

Comment (Northbadge) [reply compiler_opt/rl/corpus.py:127]: you mean the list()? So a copy is returned, to prevent mutation of the internal list

Comment (mtrofin) [reply compiler_opt/rl/corpus.py:127]: a, ok. so `get_modules_copy`? and it's not a @property then?

File: compiler_opt/rl/corpus.py [hunk 2]
```diff
@@ -86,20 +86,23 @@ def __init__(self,
                data_path: str,
                additional_flags: Tuple[str, ...] = (),
                delete_flags: Tuple[str, ...] = ()):
-    self._module_specs = _build_modulespecs_from_datapath(
-        data_path=data_path,
-        additional_flags=additional_flags,
-        delete_flags=delete_flags)
+    self.module_specs = tuple(
```
Comment (yundiqian) [inline compiler_opt/rl/corpus.py:89]: make it private and use @property to expose it instead?

**Review by yundiqian — APPROVED** [review]

**Pull request merged**